openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -749,6 +774,25 @@ def parse_smirnoff_from_source(self, source):
             A representation of a SMIRNOFF-format data structure. Begins at top-level 'SMIRNOFF' key.
 
         """
+        from openforcefield.utils import get_data_file_path
+
+        # Check whether this could be a file path. It could also be a
+        # file handler or a simple XML string.
+        if isinstance(source, str):
+            # Try first the simple path.
+            searched_dirs_paths = ['']
+            # Then try a relative file path w.r.t. an installed directory.
+            searched_dirs_paths.extend( _get_installed_offxml_dir_paths())
+            # Finally, search in openforcefield/data/.
+            # TODO: Remove this when smirnoff99Frosst 1.0.9 will be released.
+            searched_dirs_paths.append(get_data_file_path(''))
+
+            # Determine the actual path of the file.
```

Comment:
In the case that there are two files that match a given query, can we make any statement about which one will be returned? Maybe something like guaranteeing the order of entry point registration? Or can we blame any ambiguity there on user error?

Reply:
This was an issue before this PR, but it could be exacerbated by allowing FFs from external sources. I've added a TODO to keep a record of this concern.

Reply:
I'm not sure we can control it because, if it's not random, it probably depend on the order the forcefield packages will be installed. I can change the code to raise a warning or an error with a flag to load the first one to make this case a little safer.

Reply:
This is probably a complicated question, and worthy of further discussion. Like, if multiple FFs with the same name are found, but they have identical contents, maybe that's permissible? This will require some thinking about how the FF community/ecosystem will evolve. Let's open an issue to discuss, but not hold these changes up in the meantime.